### PR TITLE
TGUI: Floobot

### PR DESCRIPTION
**Pull request by Oyu07**
## What Does This PR Do
TGUI FLOORBOT
future is now old man
BOTH MERGED - READY
Waiting for #22531 and #22778 because WHY NOT

## Why It's Good For The Game
MOAR TGUI
No more WebUI

## Images of changes
image
image

## Testing
Test each option in TGUI vs WebUI

## Changelog
:cl:
tweak: Floorbot now uses TGUI
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 1]
```diff
@@ -65,33 +63,78 @@
...
+	data["locked"] = locked // controls, locked or not
+	data["noaccess"] = topic_denied(user) // does the current user have access? admins, silicons etc can still access bots with locked controls
+	data["maintpanel"] = open
+	data["on"] = on
+	data["autopatrol"] = auto_patrol
+	data["painame"] = paicard ? paicard.pai.name : null
+	data["canhack"] = canhack(user)
+	data["emagged"] = emagged // this is an int, NOT a boolean
+	data["remote_disabled"] = remote_disabled // -- STUFF BELOW HERE IS SPECIFIC TO THIS BOT
+	data["hullplating"] = autotile
```
Comment (Contrabang) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:87]: I'd recommend this for better clarity
```suggestion
	data["emagged"] = emagged // this is an int, NOT a boolean
	
	// -- STUFF BELOW HERE IS SPECIFIC TO THIS BOT
	data["remote_disabled"] = remote_disabled 
	data["hullplating"] = autotile
```

File: tgui/packages/tgui/interfaces/BotFloor.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    nag_empty,
+    magnet,
+    tiles_amount,
+  } = data;
+  return (
+    <Window resizable>
+      <Window.Content scrollable>
+        <BotStatus />
+        <Section title="Floor Settings">
+          <LabeledList>
```
Comment (Contrabang) [inline tgui/packages/tgui/interfaces/BotFloor.js:26]: I think some margin increase on the bottom would make this part look better, but thats just my opinion
```suggestion
          <LabeledList mb="3px">
```

**Review by Contrabang — APPROVED** [review]
LGTM

File: tgui/packages/tgui/interfaces/BotFloor.js (hunk 2 above)
Comment (Oyu-De) [reply tgui/packages/tgui/interfaces/BotFloor.js:25]: let me quickly test this. thx

**Comment by Oyu-De** [comment]
this branch is fucked beyond repair, it gives boostrap/python error whenever I try to edit this, I am closing this PR and will wait untill pr's depending on it comes through

**Comment by Oyu-De** [comment]
testing

**Comment by Oyu-De** [comment]
IT WORKS. ITS ALIVEEEE

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 3]
```diff
@@ -65,32 +63,65 @@
...
 
-	return dat
+/mob/living/simple_animal/bot/floorbot/show_controls(mob/user)
+	ui_interact(user)
+
+/mob/living/simple_animal/bot/floorbot/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "BotFloor", name, 500, 520)
+		ui.open()
```
Comment (Contrabang) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:74]:
```suggestion
		ui.open()
		
```

Comment (Oyu-De) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:74]: nice catch

**Comment by Contrabang** [comment]
Looks mostly fine, will review again when undrafted after  https://github.com/ParadiseSS13/Paradise/pull/22531 and https://github.com/ParadiseSS13/Paradise/pull/22778

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 4]
```diff
@@ -63,26 +63,65 @@
...
+/mob/living/simple_animal/bot/floorbot/ui_act(action, params, datum/tgui/ui)
+	if(..())
+		return
+	var/mob/user = ui.user
+	if(topic_denied(user))
+		to_chat(user, "<span class='warning'>[src]'s interface is not responding!</span>")
+		return
+	add_fingerprint(user)
+	. = TRUE
+	switch(action)
```
Comment (Contrabang) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:96]: Mostly just a style choice, but i'd prefer a blank line here
```suggestion
	. = TRUE
	
	switch(action)
```

Comment (Oyu-De) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:96]: noted, however fyi from what I can see all tgui use it without an extra line, for the sake of consistency, gonna decline this unless you got further input

**Review by farie82 — APPROVED** [review]

**Pull request merged**